### PR TITLE
Rename ./lsif to ./precise-code-intel

### PR DESCRIPTION
**Pull request by efritz**
This is the first step in a big rename away from LSIF (as an implementation detail on the server). This should not change any deployments as the lsif-server artifact has not been renamed (that will be a separate effort later).

### CODE REVIEW

File: cmd/lsif-server/Dockerfile [hunk 1]
```diff
@@ -6,11 +6,12 @@ FROM alpine:3.10@sha256:e4355b66995c96b4b468159fc5c7e3540fcef961189ca13fee877798
 RUN apk add --no-cache nodejs-current=12.4.0-r0 nodejs-npm=10.19.0-r0
 RUN npm install -g yarn@1.17.3
 
-COPY lsif/package.json lsif/yarn.lock lsif/tsconfig.json /lsif/
-RUN yarn --cwd /lsif
+# TODO - check if this copies everything
```
[the remark this answers is not in the document]
Comment (efritz) [reply cmd/lsif-server/Dockerfile]: Oh, sorry. I left this note to see if it's moving /lsif/src and busting the docker cache on all code changes. Really what I want is just the top-level files (ts, babel, jest configs, etc).

File: precise-code-intel/package.json [hunk 2]
```diff
@@ -1,9 +1,9 @@
 {
-  "name": "lsif",
-  "description": "LSIF-based precise code intelligence",
+  "name": "precise-code-intel",
+  "description": "Precise code intelligence services for Sourcegraph",
   "author": "Sourcegraph",
   "license": "MIT",
-  "version": "0.1.0",
+  "version": "0.0.0-DEVELOPMENT",
```
[the remark this answers is not in the document]
Comment (efritz) [reply precise-code-intel/package.json]: We don't publish it

[the remark this answers is not in the document]
Comment (felixfbecker) [reply precise-code-intel/package.json]: FYI you can omit them and prevent the warnings if you set `private: true` :)

[the remark this answers is not in the document]
Comment (creachadair) [reply precise-code-intel/package.json]: Nice, TIL.

File: .github/CODEOWNERS [hunk 3]
```diff
@@ -250,10 +250,9 @@ Dockerfile @sourcegraph/distribution
 /cmd/frontend/db/discussion* @slimsag
 
 # LSIF
-/lsif/ @sourcegraph/code-intel
+/cmd/lsif-server/ @sourcegraph/code-intel
```
Comment (creachadair) [inline .github/CODEOWNERS]: Good catch! Sorry I didn't think of this.

**Pull request merged**